spigot/components/src/main/kotlin/com/mineinabyss/geary/minecraft/systems/BossBarDisplaySystem.kt

```diff
@@ -0,0 +1,39 @@
+package com.mineinabyss.geary.minecraft.systems
+
+import com.mineinabyss.geary.ecs.api.systems.TickingSystem
+import com.mineinabyss.geary.ecs.engine.iteration.QueryResult
+import com.mineinabyss.geary.ecs.query.Query
+import com.mineinabyss.geary.minecraft.components.DisplayBossBar
+import org.bukkit.Bukkit
+import org.bukkit.Location
+import org.bukkit.entity.Player
+
+/**
+ * Handles displaying of boss bars.
+ */
+public object BossBarDisplaySystem : TickingSystem(interval = 10) {
+    override fun QueryResult.tick() {
+        // for each entity which has DisplayBossBar component
+        DisplayBossBarQuery.apply {
+            forEach { result: QueryResult ->
+                val displayBossBar: DisplayBossBar = result.entity.get<DisplayBossBar>() ?: return
```

Comment:
Remember the point of queries is not to have to do this, you can just do `result.bossbar`. (and if you declare the property within the system you can just call get `bossbar` as a property within this context.